lumen/ai/agents.py

```diff
@@ -255,21 +262,31 @@ class ChatAgent(Agent):
         }
     )
 
-    requires = param.List(default=[], readonly=True)
-
     async def respond(
         self,
         messages: list[Message],
+        context: TContext,
         step_title: str | None = None,
-    ) -> Any:
-        context = {"tool_context": await self._use_tools("main", messages)}
-        if "vector_metaset" not in self._memory and "source" in self._memory and "table" in self._memory:
-            source = self._memory["source"]
-            self._memory["vector_metaset"] = await get_metaset(
-                [source], [f"{source.name}{SOURCE_TABLE_SEPARATOR}{self._memory['table']}"],
-            )
-        system_prompt = await self._render_prompt("main", messages, **context)
-        return await self._stream(messages, system_prompt)
+    ) -> tuple[list[Any], dict[str, Any]]:
+        prompt_context = {"tool_context": await self._use_tools("main", messages, context)}
+        if "vector_metaset" not in context and "sql_metaset" not in context and "source" in context and "table" in context:
+            source = context["source"]
+            context["vector_metaset"] = (await get_metaset(
+                [source], [f"{source.name}{SOURCE_TABLE_SEPARATOR}{context['table']}"],
+            )).vector_metaset
+        system_prompt = await self._render_prompt("main", messages, context, **prompt_context)
+        return [await self._stream(messages, system_prompt)], {}
+
+
+class AnalystInputs(ContextModel):
+
+    data: NotRequired[Any]
+
+    source: Source
```

Comment:
data vs source might be slightly confusing here; maybe a comment or docstring would help.

(I think data is the dataframe or summary of the data table)